BK-393 time picker modal

jfplataroti opened the pull request:
Added time picker modal functionality for mobile screens. At the moment it generates a dummy collection of dates and times for the controls.
Once we connect it to the backend to get the collection of dates and times we'll adjust the bindings and selection methods.

One question, I added a package `date-fns` (I didn't want to use momentjs, it's to big) to use their `format` and `addMinutes` functions to simplify a bit the code, but I saw in your PR that you were able to format properly the dates without any library. Should I follow that approach?

![time-picker](https://user-images.githubusercontent.com/1827213/108307493-d8f84f00-717b-11eb-9f80-067751675641.png)


## Code changes
- added `TimeModal` component to show the .... time modal dialog heheh :) 
- fixed the select theming problem. It turns out that instead of adjusting the `TextInput` attribute in theme, we have to adjust `global.input`.
- added `useAvailableTime` to abstract (a bit) how we collect data from the backend. Not a big fan of the name, so let me know if you have in mind a better name :)

jfplataroti commented on packages/web/lib/theme.tsx:
```diff
@@ -64,6 +61,7 @@ const theme = {
         vertical: `${TOKENS.field.padding.vertical - 1}px`,
       },
     },
+    input: inputField,
```
added input theme into `global.input`

djoyal commented on packages/web/hooks/useAvailableTime.ts:
```diff
@@ -0,0 +1,65 @@
+  return new Array(iterate).fill(0).map((_, index) => {
+    const res = addMinutes(initialDate, timeStep * index);
+    return {
+      label: format(res, timeFormat),
```
Same as the other `Select` related comment - why not just return an array of strings (['16:00', '16:15', ... ]) instead of an object?

jfplataroti replied:
that would simplify the binding , sold! 👍

djoyal reviewed (commented):
Hey @jfplataroti, the changes look great, the UI is really clean.

Overall the code looks good. Nice use of hooks and good call on `date-fns` to help simplify date manipulation (for `addMinutes` specifically).

It looks like a bit of complexity might have been added when using the `Selects` - it looks like they depend on arrays of objects, instead of arrays of values (IE: strings). This seems to be creating a lover of overhead for interacting with the selects (there's a bunch of conversion functions). 

I think the `date-fns` is great for manipulation of the dates. I also like that the `format` is much less verbose than `toLocale...`, so I think it would be good to keep using that going forward. What do you think?

jfplataroti commented on packages/web/components/timeModal.tsx:
```diff
@@ -0,0 +1,139 @@
+  );
+};
+
+// hacky functions to convert pickup time to options,
```
Hey thanks for the feedback! :) 

> It looks like a bit of complexity might have been added when using the `Selects` - it looks like they depend on arrays of objects, instead of arrays of values (IE: strings). This seems to be creating a lover of overhead for interacting with the selects (there's a bunch of conversion functions).

Yes all those convert functions will be gone once we have the backend connected, I just wanted to create some fake data that gives a better set of values for the demo. But yes, maybe simplifying the model would be better (I'll comment on this in the suggestions provided).

> I think the `date-fns` is great for manipulation of the dates. I also like that the `format` is much less verbose than `toLocale...`, so I think it would be good to keep using that going forward. What do you think?

Sure let's keep it, I added it before realizing that we could have done the same with `toLocale...` but I agree that makes code smaller :)

jfplataroti commented on packages/web/components/timeModal.tsx:
```diff
@@ -0,0 +1,139 @@
+  const { t } = useTranslation();
+  const date = useMemo(() => new Date(), []);
+  const { hours, dates } = useAvailableHours(date);
+  const [seletedHour, setSelectedHour] = useState(
```
hahah thanks!

jfplataroti commented on packages/web/components/deliveryTabs.tsx:
```diff
@@ -11,12 +11,21 @@ import { useTranslation } from '../i18n';
+    setEstablishment,
+    setTime,
+  } = useDeliveryOptions();
+  console.log('value00000', value);
```
yes, thanks!

jfplataroti commented on packages/web/hooks/useAvailableTime.ts:
```diff
@@ -0,0 +1,65 @@
+    res.setDate(res.getDate() + index);
+    return {
+      label: format(res, dateFormat),
+      value: format(res, dateFormat),
```
Yes the `label`, `value` structure was to match the select options. At the beginning `value` was a Date, so whenever we return a value we return the date with all the data in it (date + time), but it didn't work as I expected. To set the selected value of a select component it needs a comparable value like a string or number :( 

The suggestion makes more sense than the actual code :). I'll update it to a collection of strings for the `Time` select and update the dates formats for those formats.

jfplataroti commented on packages/web/lib/theme.tsx:
```diff
@@ -48,7 +48,7 @@ const theme = {
     },
     font: {
-      family: 'FieldGrot-Regular',
+      family: 'FieldGrot',
```
updated the fontFamily to match the one declared in global.css

jfplataroti commented:
I updated the PR base on the suggestions:

- simplify `TimeModal` useStates to keep track the selected value of the fields:
  - selectedTime is a `string`
  - selectedDate is  `DateOption` -> {label: string, value: string}.
- removed uneeded convert functions used just to mock some data.

djoyal reviewed (approved):
Nice refactor!
